README.md

```diff
@@ -36,4 +42,44 @@ a `usage_point_proportional_allocator(proportion=60, edith_customers=["A", "B",
 allocation is for UP5 to receive name "A", UP2 to receive name "B", and UP3 to receive name "C". UP1 and UP4 would not
 be modified in this case.
 
-The `seed` parameter is used to seed the pseudorandom number generator, making the random allocations reproducible.
+The `seed` parameter is used to seed the pseudorandom number generator, making the random allocations reproducible. An
+example is provided in the above "Usage" section.
+
+## Line Weakener ##
+
+    from zepben.edith import line_weakener
+
+    mutator = line_weakener(
+        weakening_percentage=30,
+        use_weakest_when_necessary=False  # exclude to use weakest line type when necessary
+        match 
+    )
+
+The line weakener decreases the amp rating and modifies impedance of lines by selecting from a built-in catalogue of 
+linecodes. For each line, the applied linecode must match its phase count and whether its voltage category (HV or LV).
+
+Suppose a line has an amp rating of 300A. with a `weakening_percentage` of 30, the linecode used will have an amp rating
+of at most (100 - 30)% of 300A, i.e. 210A. Linecode "ABC4w:70ABC" would be used if the line is 3-phase LV, with an amp
+rating of 175A. Note that if there are multiple linecodes that fit the criteria, the one with the highest amp rating is
+used. If there are no linecodes that fit the criteria and `use_weakest_when_necessary` is `True`, then the
+lowest-amp-rating linecode that matches the phase and voltage criteria is used, if there is one. Otherwise, the line is
+left unmodified.
+
+## Transformer Weakener ##
+
+    from zepben.edith import transformer_weakener
+
+    mutator = transformer_weakener(
+        weakening_percentage=30,
+        use_weakest_when_necessary=False,  # exclude to use weakest transformer model when necessary
+        match_voltages=False,  # exclude to ensure transformer models match winding voltages
+    )
```

Comment:
how do we actually use the mutators?

Can you give an example of running it?

Reply:
See if the new README is clear enough

Reply:
Actually I made a function that combines multiple mutators. Or do you prefer being able to directly specify the list in the arguments for `create_synthetic_feeder`?

Reply:
The benefit of the latter is that you could get the set of changed object mrids separately:
```
changed_line_mrids, changed_tx_end_mrids = create_synthetic_feeder("fdr", mutators=[line_weakener(20), transformer_weakener(30)])
```

Reply:
I think that's a good idea however rather than returning something per mutator why don't we just allow users to pass a callback into the mutator that is called with the results?
I.e make it so all the mutators take an `on_success: Optional[Callable[[], SomeResultType]]` as an optional argument that's called with the result from the mutator.
This would allow us to have different result types for different mutators and potentially include more information in them if we see fit.